crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3479,8 +3551,7 @@ impl<'db> Type<'db> {
                                 Parameters::new([Parameter::positional_only(Some(
                                     Name::new_static("cls"),
                                 ))
-                                // TODO: type[_T]
-                                .with_annotated_type(Type::any())]),
+                                .with_annotated_type(KnownClass::Type.to_instance(db))]),
```

Comment:
I think it's fine to annotate this with `type`. We don't need the generic `type[_T]` form, as we're going to compute the return type "manually" anyway.